cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/util/AbstractAppFabricHttpHandler.java

```diff
@@ -298,6 +302,43 @@ protected void getLiveInfo(HttpRequest request, HttpResponder responder, String
     }
   }
 
+  protected void getProgramSchedules(HttpResponder responder, String namespaceId,
+                                     final String applicationId, final String programId,
+                                     ProgramType programType, Store store) {
+
+    try {
+      Id.Application appId = Id.Application.from(namespaceId, applicationId);
+      List<ProgramRecord> programRecords = listProgramsByApp(appId, programType, store);
+      if (programRecords == null) {
+        responder.sendStatus(HttpResponseStatus.NOT_FOUND);
+      } else {
+        for (ProgramRecord programRecord : programRecords) {
```

Comment:
Why is this outer for-loop necessary? Why not simply get the application spec, and iterate over the schedules only?